Validation/MuonME0Digis/src/ME0DigisValidation.cc

```diff
@@ -0,0 +1,165 @@
+//          me0_strip_digi_DeltaY[region_num][layer_num] = ibooker.book1D(histo_name_DeltaY.c_str(), histo_label_DeltaY.c_str(),100,-10,10);
+      }
+  }
+}
+
+
+ME0DigisValidation::~ME0DigisValidation() {
+}
+
+
+void ME0DigisValidation::analyze(const edm::Event& e,
+                                     const edm::EventSetup& iSetup)
+{
+ edm::ESHandle<ME0Geometry> hGeom;
+ iSetup.get<MuonGeometryRecord>().get(hGeom);
+ const ME0Geometry* ME0Geometry_ =( &*hGeom);
+  edm::Handle<edm::PSimHitContainer> ME0Hits;
+  e.getByToken(InputTagToken_, ME0Hits);
+
+  edm::Handle<ME0DigiPreRecoCollection> ME0Digis;
+  e.getByToken(InputTagToken_Digi, ME0Digis);
+
+  if (!ME0Hits.isValid() | !ME0Digis.isValid() ) {
+    edm::LogError("ME0DigisValidation") << "Cannot get ME0Hits/ME0Digis by Token simInputTagToken";
+    return ;
+  }
+
+
+
+//  for (auto hits=ME0Hits->begin(); hits!=ME0Hits->end(); hits++) {
+
+//    const ME0DetId id(hits->detUnitId());
+//    Int_t region = id.region();
+//    Int_t layer = id.layer();
+
+
+
+//    //Int_t even_odd = id.chamber()%2;
+//    if ( ME0Geometry_->idToDet(hits->detUnitId()) == nullptr) {
+//      std::cout<<"simHit did not matched with GEMGeometry."<<std::endl;
+//      continue;
+//    }
+//    const LocalPoint p0(0., 0., 0.);
+//    const GlobalPoint Gp0(ME0Geometry_->idToDet(hits->detUnitId())->surface().toGlobal(p0));
+//    const LocalPoint hitLP(hits->localPosition());
+    
+//    const GlobalPoint hitGP(ME0Geometry_->idToDet(hits->detUnitId())->surface().toGlobal(hitLP));
+//    Float_t g_r = hitGP.perp();
+//    Float_t g_x = hitGP.x();
+//    Float_t g_y = hitGP.y();
+//    Float_t g_z = hitGP.z();
+
+//    const LocalPoint hitEP(hits->entryPoint());
+
+////move these here in order to use a GEMDetId - layers, station...
+//    Float_t energyLoss = hits->energyLoss();
+//    Float_t timeOfFlight = hits->timeOfFlight();
+//    if (abs(hits-> particleType()) == 13)
+//    {
+//      timeOfFlightMuon = hits->timeOfFlight();
+//      energyLossMuon = hits->energyLoss();
+////fill histos for Muons only
+//      me0_sh_tofMu[(int)(region/2.+0.5)][layer-1]->Fill(timeOfFlightMuon);
+//      me0_sh_elossMu[(int)(region/2.+0.5)][layer-1]->Fill(energyLossMuon*1.e9);
+//    }
+    
+//      // fill hist
+//    // First, fill variable has no condition.
+//    me0_sh_zr[(int)(region/2.+0.5)][layer-1]->Fill(g_z,g_r);
+//    me0_sh_xy[(int)(region/2.+0.5)][layer-1]->Fill(g_x,g_y);
+//    me0_sh_tof[(int)(region/2.+0.5)][layer-1]->Fill(timeOfFlight);
+//    me0_sh_eloss[(int)(region/2.+0.5)][layer-1]->Fill(energyLoss*1.e9);
+
+//   }
+  for (ME0DigiPreRecoCollection::DigiRangeIterator cItr=ME0Digis->begin(); cItr!=ME0Digis->end(); cItr++) {
+    ME0DetId id = (*cItr).first;
+
+    const GeomDet* gdet = ME0Geometry_->idToDet(id);
+    if ( gdet == nullptr) {
+      std::cout<<"Getting DetId failed. Discard this gem strip hit.Maybe it comes from unmatched geometry."<<std::endl;
```

Comment:
use the message logger instead of cout (or maybe you don't need the printout at all if this is expected to be a normal condition - if its not normal, maybe you should throw and exception)